src/main/java/net/dries007/tfc/client/screen/button/KnappingButton.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.client.screen.button;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.audio.SimpleSound;
+import net.minecraft.client.audio.SoundHandler;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.client.renderer.texture.AtlasTexture;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.text.StringTextComponent;
+
+import net.minecraftforge.fml.network.PacketDistributor;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.dries007.tfc.client.TFCSounds;
+import net.dries007.tfc.network.PacketHandler;
+import net.dries007.tfc.network.ScreenButtonPacket;
+
+public class KnappingButton extends Button
+{
+    public int id;
+    private final ResourceLocation texture;
+    private final SoundEvent sound;
+
+    public KnappingButton(int id, int x, int y, int width, int height, ResourceLocation texture, SoundEvent sound)
+    {
+        super(x, y, width, height, StringTextComponent.EMPTY, KnappingButton::doPress);
+        this.id = id;
+        this.texture = texture;
+        this.sound = sound;
+    }
+
+    public static void doPress(Button widget)
```

Comment:
Rather than having this static bouncer method, nonspecific type, and `onPress` -> `doPress` delegation, it's simpler just to pass `button -> {}` in the constructor, and then override `Button#onPress`